### DISCUSSION TITLE
Recognize current session end

### DISCUSSION THREAD

**Discussion by dedonnodev** [appwrite/appwrite#2969] [Q&A] [answered]
Hello!
I searched and couldn't find a way to recognize the end of a current session. For example if an administrator deletes your current session or because your tokens have expired.

Is there a hidden method that I haven't found in the documentation or elsewhere?🙃
create an event returning from the "account" channel that otherwise says "session ended" or something?

Or, something like [this] (https://github.com/appwrite/appwrite/discussions/2482) so we don't have to subscribe to any channels.

**Comment by eldadfux** [comment, marked as answer]
You can see when the session is about to end in the session's object `expire` attribute for the exact timestamp.

In your app you can just call account.get() which is extremely optimized for fast response, if you get 401 this mean the session is not valid. Actually, any 401 response from the API would mean your user's session is over and you might want to consider to redirect the user to the login page.

**Reply by stnguyen90** [reply]
@dedonnodev can this be marked as the answer?
